pkg/api/vlabs/validate.go

```diff
@@ -111,7 +111,7 @@ func (o *OrchestratorProfile) Validate(isUpdate bool) error {
 		}
 	}
 
-	if o.OrchestratorType != Kubernetes && o.KubernetesConfig != nil && (*o.KubernetesConfig != KubernetesConfig{}) {
```

Comment:
FYI We can't do this this test anymore because `KubernetesConfig` not includes an array property. Struct literals that contain arrays can't be compared in this way.